### PR TITLE
browser: impress: introduce slide sorter context menu

### PR DESCRIPTION
**Pull request by lpranam**
added context menu entries which works out of the box

Change-Id: Ic0b598a575c4f0fb337ae79bc51cac2330d1115f

* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/control/Control.PartsPreview.js [hunk 1]
```diff
@@ -218,6 +218,37 @@ L.Control.PartsPreview = L.Control.extend({
...
+			$.contextMenu({
+				selector: '.' + imgClassName,
+				className: 'slide-preview-contextmenu',
+				items: {
+					newslide: {
+						name: _('New Slide'),
+						callback: function() { this._map.insertPage(); }
+					},
+					duplicateslide: {
+						name: _('Duplicate Slide'),
```
Comment (timar) [inline browser/src/control/Control.PartsPreview.js:231]: Please use existing translations everywhere, if possible, e.g. here `name: _UNO('.uno:DuplicateSlide', 'presentation')`.

**Review by timar — CHANGES_REQUESTED** [review]

**Review by timar — CHANGES_REQUESTED** [review]
* Right click does not select slide/page. So the menu command will have effect on the selected slide/page, and not on the slide/page where the context menu appears.
* The style (css) of this context menu is different to other context menus. E.g. the font is different.

**Review by timar — APPROVED** [review]

File: browser/src/control/Control.PartsPreview.js [hunk 2]
```diff
@@ -171,13 +172,14 @@ L.Control.PartsPreview = L.Control.extend({
 	_createPreview: function (i, hashCode, bottomBound) {
 		var frameClass = 'preview-frame ' + this.options.frameClass;
 		var frame = L.DomUtil.create('div', frameClass, this._partsPreviewCont);
-		frame.id = 'preview-frame-part-' + i;
+		frame.id = 'preview-frame-part-' + this._idNum;
```
Comment (eszkadev) [inline browser/src/control/Control.PartsPreview.js:175]: why do we need this additional `_idNum` variable?  wasn't  `i` already doing the same?
what when we switch to master page and go back? or we add/delete slide?
I think this function will be called again but index will not be reset

**Comment by lpranam** [comment]
closed in favor of #6529

**Pull request closed without merging**